# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/libreoffice/hotfix-alpha.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- libreoffice-24.8.2.1/external/pdfium/UnpackedTarball_pdfium.mk.vanilla	2024-10-17 20:38:57.849909718 +0200
+++ libreoffice-24.8.2.1/external/pdfium/UnpackedTarball_pdfium.mk	2024-10-17 20:43:25.246572617 +0200
@@ -20,6 +20,8 @@
 
 pdfium_patches += annot.patch.1
 
+pdfium_patches += alpha.patch
+
 $(eval $(call gb_UnpackedTarball_UnpackedTarball,pdfium))
 
 $(eval $(call gb_UnpackedTarball_set_tarball,pdfium,$(PDFIUM_TARBALL)))
--- libreoffice-24.8.2.1/external/pdfium/alpha.patch.vanilla	2024-10-17 20:39:44.356575737 +0200
+++ libreoffice-24.8.2.1/external/pdfium/alpha.patch	2024-10-17 20:42:57.249906351 +0200
@@ -0,0 +1,14 @@
+--- ./build/build_config.h.vanilla	2024-10-17 20:17:33.036594516 +0200
++++ ./build/build_config.h	2024-10-17 20:19:19.683259679 +0200
+@@ -370,6 +370,11 @@
+ #define ARCH_CPU_RISCV64 1
+ #define ARCH_CPU_64_BITS 1
+ #define ARCH_CPU_LITTLE_ENDIAN 1
++#elif defined(__alpha)
++#define ARCH_CPU_ALPHA_FAMILY 1
++#define ARCH_CPU_APHA1
++#define ARCH_CPU_64_BITS 1
++#define ARCH_CPU_LITTLE_ENDIAN 1
+ #else
+ #error Please add support for your architecture in build/build_config.h
+ #endif
